more_modbus/ModbusDevice.cpp

```diff
@@ -126,9 +126,28 @@ const std::vector<std::shared_ptr<RegisterGroup>>& ModbusDevice::getGroups() con
 
 const std::vector<std::shared_ptr<RegisterMapping>>& ModbusDevice::getRewritable() const
 {
+    std::lock_guard<std::mutex> lockGuard{m_rewriteMutex};
```

Comment:
If you are returning a reference than mutex won't help much as it won't sync access to the reference after this call is done. Think about returning a copy